### PR TITLE
Remove paddle.incubate.hapi.loss and reuse paddle.nn.layer.loss in high level API

### PR DESCRIPTION
**Pull request by qingqing01**
### PR types

Function optimization 

### PR changes

APIs

### Describe

Remove paddle.incubate.hapi.loss and reuse paddle.nn.layer.loss in high level API

### CODE REVIEW

**Comment by qingqing01** [comment]
```
[05:47:37]
[Step 1/1] current pr 25590 got approvals: FALSE
[05:47:37]
[Step 1/1] + APPROVALS=FALSE
[05:47:37]
[Step 1/1] ************************************
[05:47:37]
[Step 1/1] + set +x
[05:47:37]
[Step 1/1] -e It is forbidden to disable or delete the unit-test.
[05:47:37]
[Step 1/1] 
[05:47:37]
[Step 1/1] -e If you must delete it temporarily, please add it to[https://github.com/PaddlePaddle/Paddle/wiki/Temporarily-disabled-Unit-Test].
[05:47:37]
[Step 1/1] -e Then you must have one RD (kolinwei(recommended) or zhouwei25) approval for the deletion of unit-test. 
[05:47:37]
[Step 1/1] 
[05:47:37]
[Step 1/1] -e If you have any problems about deleting unit-test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/Deleting-unit-test-is-forbidden]. 
[05:47:37]
[Step 1/1] 
[05:47:37]
[Step 1/1] -e Following unit-tests are deleted in this PR: 
[05:47:37]
[Step 1/1]  -  test_loss 
[05:47:37]
[Step 1/1] 
```

@kolinwei  Please help to review for removing test_loss

**Review by heavengate — DISMISSED** [review]

**Review by kolinwei — DISMISSED** [review]
单测删除通过，python/paddle/incubate/hapi/loss.py删除导致。

**Review by LielinJiang — DISMISSED** [review]

File: python/paddle/incubate/hapi/model.py [hunk 1]
```diff
@@ -560,15 +566,16 @@ def eval_batch(self, inputs, labels=None):
...
 
-        # To be consistent with static graph
-        # return empty loss if loss_function is None
-        return ([to_numpy(l) for l in losses], metrics) \
-            if len(metrics) > 0 else [to_numpy(l) for l in losses]
+        if self.model._loss_function and len(metrics):
+            return [to_numpy(l) for l in losses], metrics
+        elif self.model._loss_function:
+            return [to_numpy(l) for l in losses]
+        elif self.model._loss_function:
```
Comment (guoshengCS) [inline python/paddle/incubate/hapi/model.py]: Should it be `else` here

File: python/paddle/incubate/hapi/metrics.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ class Accuracy(Metric):
             
         model.prepare(
             optim,
-            loss_function=CrossEntropy(average=False),
+            loss_function=CrossEntropyLoss(),
```
Comment (guoshengCS) [inline python/paddle/incubate/hapi/metrics.py]: Might it be `CrossEntropyLoss(reduction="sum")`

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/incubate/hapi/model.py [hunk 3]
```diff
@@ -1041,9 +1049,10 @@ def prepare(self,
             optimizer (Optimizer|None): Optimizer must be set in training
                 and should be a Optimizer instance. It can be None in eval
                 and test mode.
-            loss_function (Loss|None): Loss function must be set in training
-                and should be a Loss instance. It can be None when there is
-                no loss.
+            loss_function (Loss|callable function|None): Loss function can
+                be a `fluid.dygraph.Layer` instance or any callable function
+                taken the predicted values and ground truth values as input.
+                It can be None when there is no loss.
```
Comment (saxon-zh) [inline python/paddle/incubate/hapi/model.py:1073]: 涉及到API文档的修改，提交下中文API文档的PR吧

File: python/paddle/incubate/hapi/model.py (hunk 1 above)
Comment (qingqing01) [reply python/paddle/incubate/hapi/model.py]: Done. Thanks!

File: python/paddle/incubate/hapi/metrics.py (hunk 2 above)
Comment (qingqing01) [reply python/paddle/incubate/hapi/metrics.py]: Here is only an example, for shorter code, not add `reduction="sum"`

File: python/paddle/incubate/hapi/model.py (hunk 3 above)
Comment (qingqing01) [reply python/paddle/incubate/hapi/model.py:1073]: @saxon-zh  后续还需要移动目录，代码示例还需再次修改，可否最后统一修改中文API文档？

**Comment by qingqing01** [comment]
```
2020-07-28 12:54:39 0. You must have one RD (XiaoguangHu01 or lanxianghit) and one TPM (saxon-zh or jzhang533 or swtkiwi or Heeenrrry or TCChenlong) approval for the api change for the management reason of API interface.

2020-07-28 12:54:39 1. You must have one TPM (saxon-zh or jzhang533 or swtkiwi or Heeenrrry or TCChenlong) approval for the api change for the management reason of API document.
```

```
2020-07-28 13:24:03 If you must delete it temporarily, please add it to[https://github.com/PaddlePaddle/Paddle/wiki/Temporarily-disabled-Unit-Test].

2020-07-28 13:24:03 Then you must have one RD (kolinwei(recommended) or zhouwei25) approval for the deletion of unit-test. 

2020-07-28 13:24:03 

2020-07-28 13:24:03 If you have any problems about deleting unit-test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/Deleting-unit-test-is-forbidden]. 

2020-07-28 13:24:03 

2020-07-28 13:24:03 Following unit-tests are deleted in this PR: 

2020-07-28 13:24:03  -  test_loss 
```


@XiaoguangHu01 @saxon-zh @kolinwei @guoshengCS @LielinJiang @heavengate   please help to review

File: python/paddle/incubate/hapi/model.py (hunk 3 above)
Comment (saxon-zh) [reply python/paddle/incubate/hapi/model.py:1073]: 好的，那做好备案放到后面统一改吧，提前预留好时间

**Review by LielinJiang — APPROVED** [review]

**Review by guoshengCS — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by saxon-zh — APPROVED** [review]
LGTM

**Review by kolinwei — APPROVED** [review]

**Pull request merged**